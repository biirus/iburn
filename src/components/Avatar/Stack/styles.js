export default ({ spacing, colors }) => {
  return {
    /* Styles applied to root element */
    root: {
      display: "flex",
      lineHeight: 1,
    },

    /* Styles applied to the wrapper of `Avatar` element */
    avatarWrapper: {
      position: "relative",
    },

    /* Styles applied to each `Avatar` element */
    avatar: {
      border: `2px solid ${colors.n0}`,
    },
  };
};
