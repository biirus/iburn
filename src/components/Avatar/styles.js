export default ({ colors, typography, spacing, }) => {
  const size = spacing.unit * 6;

  return {
    root: {
      ...typography.h600,
      color: colors.w700a,

      display: "inline-flex",
      alignItems: "center",
      justifyContent: "center",
      alignSelf: "baseline",
      overflow: "hidden",
      width: size,
      height: size,

      backgroundColor: colors.n100,
      borderRadius: "50%",
    },

    img: {
      width: "100%",
      height: "100%",
      objectFit: "cover",
    },
  };
};
