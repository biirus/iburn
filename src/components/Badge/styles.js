export default ({ spacing, palette, colors, borderRadius }) => {
  return {
    /* Styles applied to the root element */
    root: {
      display: "inline-flex",
      padding: [[spacing.small / 2, spacing.small]],
      lineHeight: 1,
      borderRadius,
    },

    /* Styles applied to the root element if `color="default"` */
    default: {
      backgroundColor: colors.n20a,
      color: colors.n400,
    },

    /* Styles applied to the root element if `color="primary"` */
    primary: {
      backgroundColor: palette.primary,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="error"` */
    error: {
      backgroundColor: palette.error,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="success"` */
    success: {
      backgroundColor: palette.success,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="warning"` */
    warning: {
      backgroundColor: palette.warning,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="info"` */
    info: {
      backgroundColor: palette.info,
      color: palette.text.invert,
    },
  };
};
