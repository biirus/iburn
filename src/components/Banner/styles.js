export default ({
  spacing,
  palette,
  shadows,
  typography,
  zIndex,
  transitions,
}) => {
  const iconSize = spacing.unit * 3;

  return {
    root: {
      ...typography.h100,

      position: "fixed",
      top: 0,
      left: 0,
      width: "100%",
      zIndex: zIndex.flag,

      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      padding: spacing.medium,

      boxShadow: shadows.small,
      backgroundColor: palette.backgrounds.default,
      color: palette.text.primary,

      opacity: 0,
      transform: `translateY(-100%)`,

      transition: transitions.create(["opacity", "transform",], "leavingScreen"),
    },

    open: {
      opacity: 1,
      transform: `translateY(0)`,

      transition: transitions.create(
        ["opacity", "transform",],
        "enteringScreen"
      ),
    },

    warning: {
      backgroundColor: palette.messages.warning,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.warning,
      },
    },

    error: {
      backgroundColor: palette.messages.error,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.error,
      },
    },

    info: {
      backgroundColor: palette.messages.info,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.info,
      },
    },

    success: {
      backgroundColor: palette.messages.success,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.success,
      },
    },

    icon: {
      flex: `0 0 ${iconSize}px`,
      lineHeight: 0,
    },

    label: {
      flex: `0 1 auto`,
      marginLeft: spacing.small / 2,
      overflow: "hidden",
      textOverflow: "ellipsis",
      whiteSpace: "nowrap",
      lineHeight: `${iconSize}px`,
    },
  };
};
