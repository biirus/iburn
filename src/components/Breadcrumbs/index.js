import React from "react";
import PropTypes from "prop-types";
import cn from "classnames";

// components
import Typography from "components/Typography";

// enhancers
import { enhance } from "./enhance";

class Breadcrumbs extends React.Component {
  static propTypes = {
    /**
     * Class name string to be merged to root node
     */
    className: PropTypes.string,
    /**
     * [JSS](http://cssinjs.org/react-jss/?v=v8.5.1) classes object notation
     */
    classes: PropTypes.object.isRequired,
    /**
     * Array of visited pages
     */
    items: PropTypes.arrayOf(
      PropTypes.shape({
        label: PropTypes.string.isRequired,
        href: PropTypes.string.isRequired,
      })
    ),
  };

  render () {
    const { className, classes, items, theme, sheet, ...rest } = this.props;
    const classNames = cn(classes.root, className);

    if (!items) {
      return null;
    }

    return (
      <Typography {...rest} className={classNames} type="h200" component="div">
        {items.map((item, index) => (
          <span key={index} className={classes.item}>
            <a href={item.href} className={classes.link}>
              {item.label}
            </a>
          </span>
        ))}
      </Typography>
    );
  }
}

export default enhance(Breadcrumbs);
