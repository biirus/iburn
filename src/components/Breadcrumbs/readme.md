```js
<Breadcrumbs
  items={[
    { label: "Root", href: "/" },
    { label: "First child", href: "/first" }
  ]}
/>
```

### With PageHeader

```js
<PageHeader
  actions={
    <div
      style={{
        display: "grid",
        gridGap: "8px",
        gridTemplateColumns: "repeat(2, auto)"
      }}
    >
      <Button color="primary">Main action</Button>
      <Dropdown position="right" iconAfter={<Icon label="more" />}>
        <DropdownItem>Save XSLS</DropdownItem>
        <DropdownItem>Change view</DropdownItem>
      </Dropdown>
    </div>
  }
  breadcrumbs={
    <Breadcrumbs
      items={[
        { label: "Root", href: "/" },
        { label: "First child", href: "/first" }
      ]}
    />
  }
>
  Current page
</PageHeader>
```
