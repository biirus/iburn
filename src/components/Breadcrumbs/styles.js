export default theme => {
  return {
    /* Styles applied to the root element. */
    root: {
      overflow: "hidden",
      whiteSpace: "nowrap",
      textOverflow: "ellipsis",
      color: theme.palette.text.secondary,
    },

    /* Styles applied to the item's `span` element */
    item: {
      "&+$item::before": {
        content: `"/"`,
        display: "inline-block",
        width: theme.spacing.medium,
        textAlign: "center",
      },
    },

    /* Styles applied to the item's `a` element */
    link: {
      color: theme.palette.text.secondary,
      textDecoration: "none",

      "&:hover": {
        textDecoration: "underline",
      },
    },
  };
};
