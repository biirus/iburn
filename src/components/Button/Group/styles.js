export default ({ borderRadius, }) => ({
  root: {
    display: "inline-flex",
  },

  justified: {
    width: "100%",

    "& $item": {
      flex: "1 0 auto",
      width: 0,

      "& $inner": {
        width: "100%",
        textAlign: "center",
      },
    },
  },

  item: {
    flex: "1 0 auto",

    "& + $item": {
      marginLeft: 1,
    },

    "&:not(:first-child):not(:last-child) $inner": {
      borderRadius: 0,
    },

    "&:first-child $inner": {
      borderRadius: [borderRadius, 0, 0, borderRadius,],
    },

    "&:last-child $inner": {
      borderRadius: [0, borderRadius, borderRadius, 0,],
    },
  },

  inner: {},
});
