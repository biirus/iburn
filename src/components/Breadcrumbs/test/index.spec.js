import React from "react";

import Breadcrumbs from "../index";
import styles from "../styles";

const classes = getClasses(styles);
const Component = Breadcrumbs.InnerComponent;

describe("<Breadcrumbs />", () => {
  it("should merge user classes", () => {
    const wrapper = shallow(
      <Component
        classes={classes}
        className="customClassName"
        items={[{ label: "Root", href: "/", },]}
      />
    );

    expect(wrapper.prop("className")).toEqual("root customClassName");
  });

  it("should render a <Typography />", () => {
    const wrapper = shallow(
      <Component
        classes={classes}
        className="customClassName"
        items={[{ label: "Root", href: "/", },]}
      />
    );

    expect(wrapper.name()).toEqual("Jss(Typography)");
  });

  it("should return `null` if no `items` have been passed", () => {
    const wrapper = shallow(<Component classes={classes} />);
    expect(wrapper.type()).toBeNull();
  });

  it("shold pass the `rest` props to the root node", () => {
    const wrapper = shallow(
      <Component
        classes={classes}
        items={[{ label: "Root", href: "/", },]}
        dataTest="rest prop"
      />
    );

    expect(wrapper.at(0).prop("dataTest")).toEqual("rest prop");
  });

  it("should render Breadcrumbs correctly", () => {
    const wrapper = shallow(
      <Component
        classes={classes}
        items={[
          { label: "root", href: "/", },
          { label: "child", href: "/child", },
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
