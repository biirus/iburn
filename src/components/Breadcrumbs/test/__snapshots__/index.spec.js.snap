// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumbs /> should render Breadcrumbs correctly 1`] = `
<Jss(Typography)
  className="root"
  component="div"
  componentsMap={
    Object {
      "body": "div",
      "h100": "div",
      "h200": "div",
      "h300": "h5",
      "h400": "h4",
      "h500": "h3",
      "h600": "h2",
      "h700": "h1",
      "h800": "h1",
      "label": "div",
      "lead": "div",
      "paragraph": "p",
    }
  }
  type="h200"
>
  <span
    className="item"
    key="0"
  >
    <a
      className="link"
      href="/"
    >
      root
    </a>
  </span>
  <span
    className="item"
    key="1"
  >
    <a
      className="link"
      href="/child"
    >
      child
    </a>
  </span>
</Jss(Typography)>
`;
